packages/web-client/app/models/animated-workflow.ts

```diff
@@ -0,0 +1,287 @@
+  model: PostableCollection;
+  @tracked revealPointer: WorkflowPostable | undefined;
+
+  constructor(model: PostableCollection) {
+    this.model = model;
+  }
+
+  get visiblePostables(): WorkflowPostable[] {
+    let result = [] as WorkflowPostable[];
+    if (!this.revealPointer) return result;
+    for (const postable of this.model.visiblePostables) {
+      result.push(postable);
+      if (postable === this.revealPointer) {
+        break;
+      }
+    }
+    return result;
+  }
+
+  get allPostablesVisible() {
+    return (
+      this.model.allNecessaryPostablesVisible &&
+      this.model.peekAtVisiblePostables().length ===
+        this.visiblePostables.length
+    );
+  }
+
+  containsPostable(postable: WorkflowPostable) {
+    return this.model.postables.includes(postable);
+  }
+
+  resetTo(card: WorkflowPostable | undefined) {
+    if (!card || this.containsPostable(card)) this.revealPointer = card;
+  }
+
+  revealNext(): RevealResult {
+    let postables = this.model.visiblePostables;
+    let index = this.revealPointer
+      ? postables.indexOf(this.revealPointer!)
+      : -1;
+    if (index === postables.length - 1) {
+      return {
+        revealed: false,
+      };
+    }
+    this.revealPointer = postables[index + 1];
+    return {
+      revealed: true,
+      postable: this.revealPointer,
+    };
+  }
+
+  get isComplete() {
+    if (!this.model.isComplete) {
+      return false;
+    }
+    let postables = this.model.visiblePostables;
+    let index = this.revealPointer
+      ? postables.indexOf(this.revealPointer!)
+      : -1;
+    return index === postables.length - 1;
+  }
+}
+
+export default class AnimatedWorkflow {
+  model: Workflow;
+  epilogue: AnimatedPostableCollection;
+  cancelationMessages: AnimatedPostableCollection;
+  milestones: AnimatedMilestone[];
+  #listenersToCleanUp: UnbindEventListener[] = [];
+
+  constructor(model: Workflow) {
+    this.model = model;
+    this.#listenersToCleanUp.push(
+      this.model.on(
+        'visible-postables-changed',
+        this.startTestWaiter.bind(this)
+      )
+    );
+    this.#listenersToCleanUp.push(
+      this.model.on('reset-postables', this.resetPointerTo.bind(this))
+    );
+    this.epilogue = new AnimatedPostableCollection(model.epilogue);
+    this.cancelationMessages = new AnimatedPostableCollection(
+      model.cancelationMessages
+    );
+    this.milestones = model.milestones.map((m) => new AnimatedMilestone(m));
+    taskFor(this.tickerTask).perform();
+  }
+  @reads('model.name') declare name: string;
+
+  @task
+  *tickerTask() {
+    this.startTestWaiter();
+    // short timeout to prevent moving revealpointer within the same runloop
+    yield rawTimeout(10);
+
+    while (true) {
+      let result = this.revealNext();
+
+      // the last card in these things is not completed so isComplete will never
+      // be true. We check for completion by making sure all the things we need to
+      // show are shown
```

Comment:
> the last card in these things is not completed
Why is this true? Does it need to be?

Reply:
I think it doesn't have to be. We could hardcode the "Next steps" CTA to have `isComplete` be true.

Alternatively, we could also subclass `PostableCollection` and have it determine `isComplete` based on whether all necessary postables are visible.

Reply:
> I think it doesn't have to be. We could hardcode the "Next steps" CTA to have isComplete be true.

This! 👍

Reply:
Prior to this, will be changing the way that styling is applied to workflow cards to allow cards to override default width assignment/allocation by the thread. 

After making "Next steps" completed, will replace this check with `PostableCollection.isComplete`.

These changes will be made in future PRs.